connect/mirror/src/main/java/org/apache/kafka/connect/mirror/MirrorCheckpointTask.java

```diff
@@ -134,9 +138,9 @@ public String version() {
     @Override
     public List<SourceRecord> poll() throws InterruptedException {
         try {
-            long deadline = System.currentTimeMillis() + interval.toMillis();
-            while (!stopping && System.currentTimeMillis() < deadline) {
-                offsetSyncStore.update(pollTimeout);
+            if (stopping.await(pollTimeout.toMillis(), TimeUnit.MILLISECONDS)) {
```

Comment:
The condition is correct, but i agree that changing this doesn't help very much. I was mostly doing this to avoid the "busy loop" warning, but i think that isn't really worth it, especially if we can have a follow-up that actually improves the waiting behavior and eliminates that warning. Reverted.

Reply:
Ah yeah, totally right, the condition was correct. Sorry about that!